### PR TITLE
ProcessInstanceManagement SO changes

### PR DESCRIPTION
**Pull request by ivankhripunov**
Version 1.12.1.*
1. Added a new property to the Goto Activity method, 'Ignore process Instance Not Exists' which suppresses an error message if the specified process instance doesn't exist.
2. Added a new Goto Activity 2 method whoich returns a result of the operation
3. Added the 'Activity System Name' return property to the 'List Activities' method
4. Added description of the above to the ManagementAPI.adoc

### CODE REVIEW

File: K2Field.K2NE.ServiceBroker/ServiceObjects/Management API/ProcessInstanceManagementSO.cs [hunk 1]
```diff
@@ -79,12 +100,25 @@ public override List<ServiceObject> DescribeServiceObjects()
...
+        }
+        private void GotoActivity2()
+        {
+            ServiceObject serviceObject = base.ServiceBroker.Service.ServiceObjects[0];
+            serviceObject.Properties.InitResultTable();
+            DataTable results = base.ServiceBroker.ServicePackage.ResultTable;
+            _gotoActivity(results);
+        }
+
+        private void _gotoActivity(DataTable results = null)
```
Comment (cyclops1982) [inline K2Field.K2NE.ServiceBroker/ServiceObjects/Management API/ProcessInstanceManagementSO.cs:115]: Please update towards coding standard as used in the rest of the code.

File: K2Field.K2NE.ServiceBroker/ServiceObjects/Management API/ProcessInstanceManagementSO.cs [hunk 2]
```diff
@@ -24,6 +24,9 @@ public override void Execute()
                 case Constants.Methods.ProcessInstanceManagement.GotoActivity:
                     GotoActivity();
                     break;
+                case Constants.Methods.ProcessInstanceManagement.GotoActivity2:
```
Comment (cyclops1982) [inline K2Field.K2NE.ServiceBroker/ServiceObjects/Management API/ProcessInstanceManagementSO.cs:27]: If we're only ADDING properties, and providing a result. Then i think we can just add that to the existing method. I believe adding properties won't break things if those properties are not required. You can default to the previous behavior maybe?

File: Documentation/ManagementAPI.adoc [hunk 3]
```diff
@@ -23,3 +23,7 @@ RoleItems are users or groups (or SmartObjects, but those are not supported at t
 * Remove Role Item: Remove a role item from a Role.
 * List Role Items: List all the role items in a Role.
 
+=== Process Instance Management
+* Goto Activity: Executes a go to activity method. There is an input property 'Ignore process Instance Not Exists' which suppresses an error message if the specified process instance doesnt exist.
+* Goto Activity2: Executes a go to activity method and returns a result of the operation (true = successful, false = failed). There is an input property 'Ignore process Instance Not Exists' which suppresses an error message if the specified process instance doesnt exist.
```
Comment (cyclops1982) [inline Documentation/ManagementAPI.adoc:28]: What's exactly the difference between these two? Just the result?

File: K2Field.K2NE.ServiceBroker/ServiceObjects/Management API/ProcessInstanceManagementSO.cs (hunk 2 above)
Comment (ivankhripunov) [reply K2Field.K2NE.ServiceBroker/ServiceObjects/Management API/ProcessInstanceManagementSO.cs:27]: We're changing the method type as well. I'm not sure how changing from Execute to Read will affect the current usages. That's why I added a new method.

**Pull request merged**